rules/library.bzl

```diff
@@ -267,10 +296,37 @@ def _xcframework_slice(*, xcframework_name, identifier, platform, platform_varia
             deps = [],
             tags = _MANUAL,
         )
+        imported_framework_name = _find_imported_framework_name(import_headers)
+        vfs_framework_name = imported_framework_name if imported_framework_name else xcframework_name
+        framework_vfs_overlay(
+            name = resolved_target_name_vfs_overlay,
+            framework_name = vfs_framework_name,
+            modulemap = import_module_map,
+            swiftmodules = import_swiftmodules,
+            hdrs = import_headers,
+            tags = _MANUAL,
+            extra_search_paths = xcframework_name,
+        )
     elif (linkage, packaging) == ("static", "library"):
+        # For a static library the `path` is the path to the library (`.a`) itself.
+        # We need the path to the top-level slice directory so we can find headers, modulemaps and swiftmodules.
+        slice_dir = paths.dirname(path)
+        import_headers, import_module_map, import_swiftmodules = _xcframework_slice_imports(slice_dir)
         native.objc_import(
             name = resolved_target_name,
             archives = [path],
+            hdrs = import_headers,
+            includes = ["%s/Headers" % slice_dir],
+            tags = _MANUAL,
+        )
+        imported_framework_name = _find_imported_xcframework_name(import_headers)
+        vfs_framework_name = imported_framework_name if imported_framework_name else xcframework_name
+        framework_vfs_overlay(
+            name = resolved_target_name_vfs_overlay,
+            framework_name = vfs_framework_name,
+            modulemap = import_module_map,
+            swiftmodules = import_swiftmodules,
+            hdrs = import_headers,
             tags = _MANUAL,
         )
```

Comment:
The only changes here are:
 - The `framework_name` is now correct when using the new `_find_imported_xcframework_name` 
 - I removed `extra_search_paths` to copy the way we create the VFS overlay for normal static libraries. If we pass this parameter then importing fails for VFS, removing it fixes the issue